apollo-router/src/configuration/mod.rs

```diff
@@ -1255,3 +1265,16 @@ fn default_graphql_path() -> String {
 fn default_graphql_introspection() -> bool {
     false
 }
+
+/// Configuration for Batching
+#[derive(Debug, Default, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+#[serde(default)]
+pub(crate) struct Batching {
+    /// Activates Batching (disabled by default)
+    pub(crate) enabled: bool,
+
+    /// Batching mode (batch_http_link is supported)
+    #[serde(default)]
+    pub(crate) mode: String,
```

Comment:
Why not using an enum here ?

Reply:
We don't know what future versions might be added, or removed, so I'm just using a String to keep it open.